stdlib/float.ml

```diff
@@ -73,10 +84,20 @@ external sinh : float -> float = "caml_sinh_float" "sinh"
   [@@unboxed] [@@noalloc]
 external tanh : float -> float = "caml_tanh_float" "tanh"
   [@@unboxed] [@@noalloc]
+external trunc : float -> float = "caml_trunc_float" "caml_trunc"
+  [@@unboxed] [@@noalloc]
```

Comment:
caml_trunc_float calls `caml_copy_double` which allocates -- so the `noalloc` here is unsafe isn't it?

Reply:
I do not think so.  Allocation is only for the bytecode, not for native code.  (`tanh` and other functions are implemented the same way.)  I agree that the manual could be clearer.

Reply:
Oh I see -- the noalloc only applies to the native code version of the function. That is confusing indeed!